### ISSUE TITLE
Add Support For Steelseries Rival 3 Wireless

### ISSUE THREAD

**Issue by ntr09** [libratbag/libratbag#1111]
I've been tinkering with device files and cannot quite get it to work. With the device file I created for it, it is recognized under Piper but it returns "Something went wrong. The device has been reset to a previous state." when clicking "Apply".

I will be glad to offer my assistance. 

Thanks!

**Comment by duck-rh** [comment]
Quack,

I've got the wired version and I guess it should work identically so adding to this issue.

I've tried with this device configuration:
```
[Device]
Name=SteelSeries Rival 3
DeviceMatch=usb:1038:1824
Driver=steelseries

[Driver/steelseries]
DeviceVersion=2
Buttons=6
Leds=3
DpiRange=100:8500@100
MacroLength=1
```

The mouse image is missing but the info are displayed. I cannot update anything with the same error as @ntr09 . Changing DeviceVersion to 3 did not help either. ratbagd does not seem to log anything.

Using ratbagctl I can get information and change values without errors but nothing change on the mouse. for eg changing dpi has not effect, and changing it using the button on the mouse does not change the value reported by the tool. Again trying DeviceVersion 3 did not help.

**Comment by ntr09** [comment]
There are some minor differences though, @duck-rh. The Wireless's DPI is 100-18,000 and it has only 1 LED. Shouldn't be a big deal though.

**Comment by amfern** [comment]
Yes, this would require to reverse engineer their new protocol. 
@Usul75 please provide us with the version of ratbagctl you are using `ratbagctl --version`

**Comment by Usul75** [comment]
0.16

**Comment by tomaszg7** [comment]
I also have a bad luck to have this mouse. In my case I got more meaningful output of `lshid` for one of the devices: https://gist.github.com/tomaszg7/ef4901e4d416f6375fd5359ee092f8c3

I'm also using `libratbag` 0.16. After adding the device file from above, it is detected by `ratbagctl`, but I can't seem to change anything. Commands silently fail.

```
# ratbagctl list
singing-gundi:       SteelSeries SteelSeries Rival 3 Wireless

# ratbagctl singing-gundi info
singing-gundi - SteelSeries SteelSeries Rival 3 Wireless
             Model: usb:1038:1830:0
 Number of Buttons: 6
    Number of Leds: 3
Number of Profiles: 1
Profile 0: (active)
  Name: n/a
  Report Rate: 1000Hz
  Resolutions:
    0: 800dpi (active) (default)
    1: 1600dpi
  Button: 0 is mapped to 'button 1'
  Button: 1 is mapped to 'button 2'
  Button: 2 is mapped to 'button 3'
  Button: 3 is mapped to 'button 4'
  Button: 4 is mapped to 'button 5'
  Button: 5 is mapped to 'resolution-cycle-up'
  LED: 0, depth: rgb, mode: on, color: 0000ff
  LED: 1, depth: rgb, mode: on, color: 0000ff
  LED: 2, depth: rgb, mode: on, color: 0000ff

# ratbagctl singing-gundi resolution active get
0
# ratbagctl singing-gundi resolution active set 1
# ratbagctl singing-gundi resolution active get
1
```

Similarly with LED modes and colors. I can change it and output of `info` gets updated but that doesn't reflect the state of hardware. Let me know if I should do some more tests.

**Comment by TDGalea** [comment]
Another user with the Rival 3 Wirelsss here - first time finding out about libratbag/piper.

The device file posted by duck-rh does have it recognized, and as far as I can tell, after changing it to have just one LED (wireless), it seems to be pulling in the correct colour configured in Windows. However the device continues to use SteelSeries orange, with the whole "fade out when moving" effect, and attempting to change any setting at all fails.

Willing to provide any info that might help where possible.

**Comment by tomaszg7** [comment]
There's some initial support for this mouse in `rivalcfg` https://github.com/flozz/rivalcfg/issues/146 No leds yet but changing sensitivity and polling rate seem to work nicely.

**Comment by DumbledoreCalrissian** [comment]
Another Rival 3 wireless user here.
Willing to give any input that might help adding support.

**Comment by kellerman** [comment]
Bought Rival 3 non-wireless. Mouse isn't recognized. What can I help?

**Comment by clari7744** [comment]
Adding in mine as well; I got a SteelSeries Rival 3 Wireless and I can't get it to show. It appears in `lsusb`, `lshid` etc, but is not recognized by `ratbagctl` or https://github.com/libratbag/piper.

**Comment by meowseverywhere** [comment]
Trying to add my Steel Series Rival 3 wireless mouse and I tried the stuff above but it appears to not work. Any help would be appreciated
